include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -353,8 +354,14 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 				// Needed to get the vector deserialization automatically
 				SEntry() {}
 
-				inline SEntry(SEntry&) = default;
-				inline SEntry& operator=(SEntry&) = default;
+				// Making the copy constructor deep-copy
+				inline SEntry(const SEntry& other) 
+					: mainFileContents(other.mainFileContents), compilerArgs(other.compilerArgs), hash(other.hash), lookupHash(other.lookupHash), 
+					  dependencies(other.dependencies), value(core::smart_refctd_ptr_static_cast<ICPUShader, IAsset>(other.value->clone())) {}
+				
+				inline SEntry& operator=(const SEntry& other) {
+					*this = SEntry(other);
+				}
```

Comment:
bad bad bad idea to have deep copy on everything.

You shouldn't deep copy the `shader` its a shared pointer, its okay to share a reference
P.S. `value` could actually be `core::smart_refctd_ptr<const ICPUShader>` cause its readonly

if you truly need then make

Reply:
This was to be used in the cache clone. I could alternatively call this `SEntry::clone()` instead of making it the copy constructor. 

When cloning then I can just +1 the refctd shader then, right? No need to deep copy that one

Reply:
Either have a `clone()` method or a non-public copy ctor, because you can accidentally end up with lots of redundant copies because of how C++ syntax works https://github.com/Devsh-Graphics-Programming/Nabla/pull/676#discussion_r1580122994

You don't +1 manually, every time you create a `smart_refctd_ptr` the refcount increases, works exactly like `std::shared_ptr` for a class `T` that inherits from `std::enable_shared_from_this<T>`

Reply:
still outstanding